### PR TITLE
[11699 PART 4] Use queue builder to build attorney queues from config

### PR DESCRIPTION
**Pull request by hschallhorn**
Final PR of #11699

### Description
Use queue builder introduced in #13578 to build attorney queues from config. As a followup to #11698 and a precursor to the Grand Unified Queue #10032, offload the specific queue configuration for attorneys to the back end

## Acceptance Criteria
- [ ] The queue for attorneys is built from queue config rather than front end logic

### Attorney Queue
1. Go to BVASCASPER1's queue
2. Everything still works
3. UI matches master
### Colocated Queue
1. Go to CSS_ID1's queue
2. Everything still works
3. UI matches master

### Attorney queues
 - [ ] None, other than
   - [ ] Note that the only "change" in the UI is the days waiting column. These red bars won't actually show up in the user's queue unless they have put their attorney task on hold and the hold has expired. There are currently no attorney tasks on a timed hold in production, let alone any with expired holds. This is not a change that users will see.
   - [ ] Copy on the on hold tab description is updated

Master|Changes
---|---
Screen Shot 2020-03-02 at 2 56 57 PM|Screen Shot 2020-03-02 at 4 51 20 PM
Screen Shot 2020-03-02 at 2 57 04 PM|Screen Shot 2020-03-02 at 4 51 26 PM
Screen Shot 2020-03-02 at 2 57 16 PM|Screen Shot 2020-03-02 at 4 51 41 PM

### Colocated Queues
- [ ] "Days" in the days waiting tab

Master|Changes
---|---
Screen Shot 2020-03-02 at 4 47 04 PM|Screen Shot 2020-03-02 at 4 47 04 PM
Screen Shot 2020-03-02 at 4 46 31 PM|Screen Shot 2020-03-02 at 4 47 11 PM
Screen Shot 2020-03-02 at 4 46 37 PM|Screen Shot 2020-03-02 at 4 47 17 PM

[_What is this stacked pr nonsense?_](https://unhashable.com/stacked-pull-requests-keeping-github-diffs-small/)

### CODE REVIEW

File: client/COPY.json [hunk 1]
```diff
@@ -109,8 +109,6 @@
   "ATTORNEY_QUEUE_TABLE_SUCCESS_MESSAGE_DETAIL": "If you made a mistake, please email your judge to resolve the issue.",
   "ATTORNEY_QUEUE_TABLE_TASK_NEEDS_ASSIGNMENT_ERROR_MESSAGE": "Please ask your judge to assign this case to you in DAS",
   "ATTORNEY_QUEUE_TABLE_TASK_NO_DOCUMENTS_READER_LINK": "View in Reader",
-  "ATTORNEY_QUEUE_PAGE_ASSIGNED_TASKS_DESCRIPTION": "Cases assigned to you.",
-  "ATTORNEY_QUEUE_PAGE_ON_HOLD_TASKS_DESCRIPTION": "When a hold is complete, cases here will automatically return to the Assigned tab.",
```
Comment (hschallhorn) [inline client/COPY.json]: Now unused as queue config uses `USER_QUEUE_PAGE_ON_HOLD_TASKS_DESCRIPTION` and `USER_QUEUE_PAGE_NEW_TASKS_DESCRIPTION`

File: client/app/queue/components/TaskTableColumns.jsx [hunk 2]
```diff
@@ -205,7 +205,7 @@ export const daysWaitingColumn = (requireDasRecord) => {
     valueFunction: (task) => {
       return <React.Fragment>
         <span className={taskHasCompletedHold(task) ? 'cf-red-text' : ''}>{moment().startOf('day').
-          diff(moment(task.assignedOn), 'days')}</span>
+          diff(moment(task.assignedOn), 'days')} days</span>
```
Comment (hschallhorn) [inline client/app/queue/components/TaskTableColumns.jsx]: I can take or leave this change. This keeps "days" in the days waiting column for attorneys but introduces it into colocated queues

File: client/app/queue/AttorneyTaskListView.jsx [hunk 3]
```diff
@@ -147,20 +114,23 @@ const mapDispatchToProps = (dispatch) => ({
...
-    includeDocketNumber
-    includeIssueCount
-    includeDueDate
-    includeReaderLink
-    requireDasRecord
-    tasks={tasks}
-    includeNewDocsIcon={includeNewDocsIcon}
-    useOnHoldDate={useOnHoldDate}
-  />
-</React.Fragment>;
```
Comment (hschallhorn) [inline client/app/queue/AttorneyTaskListView.jsx]: Column selection is handled in [queue builder](https://github.com/department-of-veterans-affairs/caseflow/pull/13578/files#diff-23e953264b3a42367eaf048934e912d2R37) along with [back end config](https://github.com/department-of-veterans-affairs/caseflow/pull/13579/files#diff-68a8cf56cbc30400ee698ea2242eb54eR55)

File: client/app/queue/AttorneyTaskListView.jsx [hunk 4]
```diff
@@ -55,59 +53,27 @@ class AttorneyTaskListView extends React.PureComponent {
...
-        label: COPY.QUEUE_PAGE_COMPLETE_TAB_TITLE,
-        page: <TaskTableTab
-          description={COPY.QUEUE_PAGE_COMPLETE_TASKS_DESCRIPTION}
-          tasks={this.props.completedTasks}
-          includeNewDocsIcon={false}
-          useOnHoldDate={false}
-        />
-      }
-    ];
-
```
Comment (hschallhorn) [inline client/app/queue/AttorneyTaskListView.jsx]: Tab selection is handled by [queue builder](https://github.com/department-of-veterans-affairs/caseflow/pull/13578/files#diff-23e953264b3a42367eaf048934e912d2R27) along with [back end config](https://github.com/department-of-veterans-affairs/caseflow/blob/43562ed9e75b14c6f949802521da9df6a4214c2b/app/models/user.rb#L319)

**Comment by yoomlam** [comment]
Not sure why I'm seeing duplicates in the On hold tab after assigning an admin task:
image

Note that one has a red bar and the other doesn't.
```
                                      ┌─────────────────────────────────────────────────────────────────────┐
LegacyAppeal 149 (legacy) ─────────── │ ID   │ STATUS   │ ASGN_BY     │ ASGN_TO   │ UPDATED_AT              │
├── RootTask                          │ 1174 │ assigned │             │ Bva       │ 2020-03-03 16:59:26 UTC │
└── PoaClarificationColocatedTask     │ 1177 │ on_hold  │ BVASCASPER1 │ Colocated │ 2020-03-03 17:01:05 UTC │
    └── PoaClarificationColocatedTask │ 1178 │ assigned │ BVASCASPER1 │ CSS_ID1   │ 2020-03-03 17:01:05 UTC │
                                      └─────────────────────────────────────────────────────────────────────┘
```

File: spec/feature/queue/task_queue_spec.rb [hunk 5]
```diff
@@ -143,20 +143,20 @@
...
+      expect(page).to have_content(COPY::USER_QUEUE_PAGE_TABLE_TITLE)
       expect(page).to have_content(format(COPY::QUEUE_PAGE_ASSIGNED_TAB_TITLE, vacols_tasks.length))
       expect(page).to have_content(format(COPY::QUEUE_PAGE_ON_HOLD_TAB_TITLE, attorney_on_hold_task_count))
       expect(page).to have_content(COPY::QUEUE_PAGE_COMPLETE_TAB_TITLE)
     end
 
     it "shows the right number of cases in each tab" do
       # Assigned tab
-      expect(page).to have_content(COPY::ATTORNEY_QUEUE_PAGE_ASSIGNED_TASKS_DESCRIPTION)
+      expect(page).to have_content(COPY::USER_QUEUE_PAGE_NEW_TASKS_DESCRIPTION)
```
Comment (yoomlam) [inline spec/feature/queue/task_queue_spec.rb]: I would name this `USER_QUEUE_PAGE_ASSIGNED_TASKS_DESCRIPTION` to parallel `USER_QUEUE_PAGE_ON_HOLD_TASKS_DESCRIPTION` and `QUEUE_PAGE_COMPLETE_TASKS_DESCRIPTION`.

File: client/app/queue/components/TaskTableColumns.jsx (hunk 2 above)
Comment (yoomlam) [reply client/app/queue/components/TaskTableColumns.jsx]: I like consistency.

Comment (yoomlam) [reply client/app/queue/components/TaskTableColumns.jsx]: Do you want to use `${pluralize('day', diff)}`?

**Comment by hschallhorn** [comment]
> Not sure why I'm seeing duplicates in the On hold tab after assigning an admin task:

@yoomlam Looks like this happens on master as well
Screen Shot 2020-03-03 at 12 26 48 PM
Filing a ticket!

**Comment by yoomlam** [comment]
For the "On hold" tab of CSS_ID1, the "Days on Hold" column value actually says "N of M":

image

which is different compared to your screenshots.

**Comment by yoomlam** [comment]
> which is different compared to your screenshots.

The syntax for the "Days on Hold" column values depend on the type of task that is causing the hold (`TimedHoldTask` vs others):

image

**Pull request merged**